### PR TITLE
WSGI adapter

### PR DESCRIPTION
**Pull request by jmacarthur**
I've made an example WSGI adapter so I can run an example of Doorstop on one of our servers; it might be useful for others.

### CODE REVIEW

File: docs/reference/webserver.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# Using the Doorstop web front-end
+
+## Conventional bottle webserver
+
+Doorstop can be run as a standalone web server by running
+`python3 doorstop/server/main.py`.  It will use the current working
```
Comment (jacebrowning) [inline docs/reference/webserver.md]: This is actually exposed as a command-line client, `doorstop-server`. I think that should be the recommended approach here.

File: docs/reference/webserver.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+directory as the document source by default. You may need to install
+the [Python Bottle framework](https://bottlepy.org/docs/dev/), which
+can be done with `pip install bottle` or `apt install python-bottle`
```
Comment (jacebrowning) [inline docs/reference/webserver.md]: Bottle will already be installed with `pip install doorstop`.

**Comment by jacebrowning** [comment]
@jmacarthur I tend to prefer keeping the history in tact once a review has started.

FYI: The linter is reporting an unused variable.

**Comment by jmacarthur** [comment]
Hmm, yes, I thought root had more uses in that function but apparently it doesn't. I'll have another go at that.

**Comment by jmacarthur** [comment]
Fixed in this branch; squashed version in jmac/wsgi4 if you'd prefer that.

**Pull request merged**